content/news/posts/2025/PSDevCon2024testingandanalyzingmodulesautomatically.md

```diff
@@ -0,0 +1,513 @@
+
+You can find the sources for the module created in this article on the following GitHub repository: PrestaEdit/modulenine. Jonathan, the floor is now yours!
+
+# Preamble
+
+When developing a module, it is very often created on a single instance and primarily on a more recent version of PrestaShop. 
+
+However, the users of these modules are not always on the same platform. 
+
+Over the course of both minor and major versions, PrestaShop introduces code changes that can impact the proper functioning of the module.
+
+Nevertheless, manually testing on each version of PrestaShop and with each batch of new methods is tedious.
+
+Performing these tests at the end of development can also prove more complicated than expected, especially if an entire section of code is not backward compatible.
+
+Let us then see how to test the code of your module automatically for compatibility across all versions.
+
+
+# PhpStan
+### Introduction
+
+[PHPStan](https://phpstan.org/) is a static code analysis tool specifically for PHP. 
+
+It analyzes your code for errors and potential issues, helping you improve the quality of your code and avoid common mistakes. 
+
+At the PrestaShop level, you can already use PhpStan outside of this project. For that, see the [official documentation](https://devdocs.prestashop-project.org/1.7/modules/testing/advanced-checks/#static-analysis).
+
+### Configuration
+
+Before getting to the matter, let's first explore the necessary setup.
+
+We are going to create two shell script files.
+
+In our example, we use the module name directly in the file, without using a variable.
+
+*About this topic, you could pass the module name directly in the GitHub workflow by using the repository name. This will not be covered in this article.*
+
+/tests/phpstan.sh
+
+
+```
+#!/bin/bash
+PS_VERSION=$1
+
+set -e
+
+# Docker images prestashop/prestashop may be used, even if the shop remains uninstalled
+echo "Pull PrestaShop files (Tag ${PS_VERSION})"
+
+docker rm -f temp-ps || true
+docker volume rm -f ps-volume || true
+
+docker run -tid --rm -v ps-volume:/var/www/html --name temp-ps prestashop/prestashop:$PS_VERSION
+
+# Run a container for PHPStan, having access to the module content and PrestaShop sources.
+# This tool is outside the composer.json because of the compatibility with PHP 5.6
+echo "Run PHPStan using phpstan-${PS_VERSION}.neon file"
+
+docker run --rm --volumes-from temp-ps \
+       -v $PWD:/var/www/html/modules/modulenine \
+       -e _PS_ROOT_DIR_=/var/www/html \
+       --workdir=/var/www/html/modules/modulenine phpstan/phpstan:0.12.54 \
+       analyse \
+       --configuration=/var/www/html/modules/modulenine/tests/phpstan/phpstan-$PS_VERSION.neon
+```
+
+
+/tests/phpstan-v9.sh
+
+
+This is necessary to enable the use of the registry prestaedit/prestashop instead of prestashop/prestashop, as the Docker image for PrestaShop 9 will only be available upon its official release.
```

Comment:
```suggestion
This is necessary to enable the prestaedit/prestashop registry use instead of prestashop/prestashop, as the Docker image for PrestaShop 9 will only be available upon its official release.
```